lib/upload.ts

```diff
@@ -0,0 +1,215 @@
+'use server'
```

Comment:
Not sure you need this one: the latest NextJS (the one we are using) is server-first by default.

```suggestion
```

Reply:
We tried this and got an error. I think it's mandatory for server actions: https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#convention. But you're right that all components are server components by default unless `use client` directive is present